### PR TITLE
test parent hash validation

### PR DESCRIPTION
**Pull request by gurukamath**
### What was wrong?
Test to validate the parent hash of a block was missing

Related to Issue #387

### How was it fixed?
Added an additional assert while testing state transition (frontier and homestead)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/frontier/blockchain_st_test_helpers.py [hunk 1]
```diff
@@ -76,6 +76,13 @@ def run_blockchain_st_test(
 
 def add_blocks_to_chain(chain: BlockChain, test_data: Dict[str, Any]) -> None:
     for idx, block in enumerate(test_data["blocks"]):
+        if idx == 0:
```
Comment (voith) [inline tests/frontier/blockchain_st_test_helpers.py:79]: These assert statements basically check that the test data is correct. This doesn't ensure that the `spec.py` actually validates the parent hash.
What we should be doing is writing a unit test that provides an invalid parent hash and checks that test reverts with a validation error.

File: tests/frontier/blockchain_st_test_helpers.py [hunk 2]
```diff
@@ -76,6 +76,13 @@ def run_blockchain_st_test(
...
+            assert test_data["genesis_header_hash"] == block.header.parent_hash
```
Comment (voith) [inline tests/frontier/blockchain_st_test_helpers.py:80]: These assert statements basically check that the test data is correct. This doesn't ensure that the `spec.py` actually validates the parent hash.
What we should be doing is writing a unit test that provides an invalid parent hash and checks that test reverts with a validation error.

Comment (gurukamath) [reply tests/frontier/blockchain_st_test_helpers.py:80]: @voith Thank you for the input. I have a few questions here.

I see that the invalid blocks test cases are already present under tests/fixtures/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest/

I don't think any of these test cases are implemented yet (at least I couldn't find the implementations). Is that right? If I've missed it, could you please point me to the implementation?

If they have not been implemented yet, please suggest the file where adding these tests would be most appropriate. (test_state_transition.py? or maybe a new file?)

Comment (voith) [reply tests/frontier/blockchain_st_test_helpers.py:80]: > I see that the invalid blocks test cases are already present under tests/fixtures/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest/


Some of those tests were added sometime back and were accidentally removed by me. Please refer to #415 for more context. You can add them back by referring to the commit I've posted in that issue. 
However, only [bcUncleTest](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcUncleTest)  and [bcUncleHeaderValidity](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcUncleHeaderValidity) were tested earlier. You probably need [bcInvalidHeaderTest](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest) in order to close #387

Comment (gurukamath) [reply tests/frontier/blockchain_st_test_helpers.py]: @voith Accidentally happened to remove the repo from which I had raised the PR. Have opened a new PR with updates to include the relevant unit tests. Please review

https://github.com/ethereum/execution-specs/pull/416

**Pull request closed without merging**